### ISSUE TITLE
Scapy breaks on import if 8-byte Firewire link layer address exists

### ISSUE THREAD

**Issue by samyk** [secdev/scapy#3350]
### Brief description

Just importing scapy triggers str2mac() to be called which expects all link layer addresses to be 6 bytes while Firewire addresses are 8 bytes per IEEE 1394.

```
>>> from scapy.all import IP
WARNING: No IPv4 address found on p2p0 !
WARNING: No IPv4 address found on en1 !
WARNING: more No IPv4 address found on en2 !
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/samy/.miniconda3/lib/python3.8/site-packages/scapy/all.py", line 16, in <module>
    from scapy.arch import *
  File "/Users/samy/.miniconda3/lib/python3.8/site-packages/scapy/arch/__init__.py", line 125, in <module>
    conf.use_bpf = True
  File "/Users/samy/.miniconda3/lib/python3.8/site-packages/scapy/config.py", line 114, in __set__
    val = self.hook(self.name, val, old, *self.args, **self.kargs)
  File "/Users/samy/.miniconda3/lib/python3.8/site-packages/scapy/config.py", line 651, in _socket_changer
    _set_conf_sockets()
  File "/Users/samy/.miniconda3/lib/python3.8/site-packages/scapy/config.py", line 609, in _set_conf_sockets
    conf.ifaces.reload()
  File "/Users/samy/.miniconda3/lib/python3.8/site-packages/scapy/interfaces.py", line 238, in reload
    self._reload_provs()
  File "/Users/samy/.miniconda3/lib/python3.8/site-packages/scapy/interfaces.py", line 234, in _reload_provs
    self._load(prov.reload(), prov)
  File "/Users/samy/.miniconda3/lib/python3.8/site-packages/scapy/interfaces.py", line 53, in reload
    return self.load()
  File "/Users/samy/.miniconda3/lib/python3.8/site-packages/scapy/arch/bpf/core.py", line 224, in load
    mac = scapy.utils.str2mac(get_if_raw_hwaddr(ifname)[1])
  File "/Users/samy/.miniconda3/lib/python3.8/site-packages/scapy/utils.py", line 579, in str2mac
    return ("%02x:" * 6)[:-1] % tuple(map(ord, s))
TypeError: not all arguments converted during string formatting
```

Here's the ifconfig interface triggering the issue on macOS 10.14.6:
```sh
ifconfig fw0
fw0: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 4078
	lladdr ec:1a:59:7b:a0:71:55:73
	nd6 options=201<PERFORMNUD,DAD>
	media: autoselect <full-duplex>
	status: inactive
```

### Scapy version

2.4.5

### Python version

3.8.3

### Operating system

macOS 10.14.6

### How to reproduce

If you have a Firewire interface, `from scapy.all import IP` will trigger the issue.

**Comment by samyk** [comment]
PR [submitted](https://github.com/secdev/scapy/pull/3351), requires review

**Comment by guedou** [comment]
Fixed by #3351
